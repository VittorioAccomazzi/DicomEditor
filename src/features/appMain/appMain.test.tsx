
import React from 'react';
import { cleanup, render } from '@testing-library/react'
import * as reactDropzone from "react-dropzone"
import AppMain from './appMain'

beforeEach(() => {
  // setup a DOM element as a render target
});

afterEach(()=>{
  cleanup()
  jest.clearAllMocks()
})

test('render the main app',()=>{
    const {container} =  render(<AppMain/>)
    expect(container.innerHTML).toMatchSnapshot()
})

test('check correct message displayed',async ()=>{
    render(<AppMain/>)

    const activeDiv = document.querySelector('[test-id=drop-active-div]')!
    const inactivDiv= document.querySelector('[test-id=drop-inactive-div]')!

    expect(activeDiv.className).toContain('noDisplay') 
    expect(inactivDiv.className).toContain('drop')
})

test('check on drag the correct message is displayed',async()=>{

  const orguse = reactDropzone.useDropzone
  jest.spyOn(reactDropzone,'useDropzone').mockImplementation(()=>({ ...orguse(), isDragActive : true}))

  render(<AppMain/>)

  const activeDiv = document.querySelector('[test-id=drop-active-div]')!
  const inactivDiv= document.querySelector('[test-id=drop-inactive-div]')!

  expect(activeDiv.className).toContain('drop') 
  expect(inactivDiv.className).toContain('noDisplay')
})
