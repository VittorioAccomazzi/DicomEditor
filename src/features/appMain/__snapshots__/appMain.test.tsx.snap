// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the main app 1`] = `"<div class=\\"makeStyles-root-1\\"><div tabindex=\\"0\\" class=\\"makeStyles-root-1\\" test-id=\\"drop-div\\"><div class=\\"makeStyles-noDisplay-3\\" test-id=\\"drop-active-div\\"><svg class=\\"MuiSvgIcon-root AddIcn\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z\\"></path></svg><p class=\\"MuiTypography-root text MuiTypography-body1\\">Drop the file here.</p></div><div class=\\"makeStyles-drop-2\\" test-id=\\"drop-inactive-div\\"><input multiple=\\"\\" type=\\"file\\" style=\\"display: none;\\" autocomplete=\\"off\\" tabindex=\\"-1\\"><svg class=\\"MuiSvgIcon-root AddIcn\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z\\"></path></svg><p class=\\"MuiTypography-root text MuiTypography-body1\\">Drag and Drop here your DICOM files or folders here.</p><p class=\\"MuiTypography-root text MuiTypography-body1\\">or click here to select the files.</p></div></div></div>"`;
